### PR TITLE
Implement Search API using Index to search deals

### PR DESCRIPTION
**Pull request by limli**

### CODE REVIEW

File: src/main/java/com/google/step/servlets/DealSearchServlet.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  /**
+   * Searches for deals with certain description and tags.
+   *
+   * <p>Request format: ?q=my+query&tags=123,456, where q is the query string, and tags is a comma
+   * separated list of tag IDs
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("search deal");
```
Comment (jyx11011) [inline src/main/java/com/google/step/servlets/DealSearchServlet.java]: Remove println:)

File: src/main/java/com/google/step/servlets/DealSearchServlet.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    String query = request.getParameter("q");
```
Comment (jyx11011) [inline src/main/java/com/google/step/servlets/DealSearchServlet.java]: Rename this parameter to a more descriptive name?

**Review by jyx11011 — DISMISSED** [review]

File: src/main/java/com/google/step/servlets/DealSearchServlet.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  private final DealSearchManager manager;
+
+  public DealSearchServlet() {
+    manager = new DealSearchManagerIndex();
+  }
+
+  /**
+   * Searches for deals with certain description and tags.
+   *
+   * <p>Request format: ?q=my+query&tags=123,456, where q is the query string, and tags is a comma
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/servlets/DealSearchServlet.java]: should this be `?query=my+...`?

File: src/main/java/com/google/step/datamanager/DealSearchManagerIndex.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import com.google.appengine.api.search.Results;
+import com.google.appengine.api.search.ScoredDocument;
+import com.google.appengine.api.search.SearchServiceFactory;
+import com.google.step.model.Deal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DealSearchManagerIndex implements DealSearchManager {
+  Index index;
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/datamanager/DealSearchManagerIndex.java]: private final

File: src/main/java/com/google/step/datamanager/DealManagerDatastore.java [hunk 5]
```diff
@@ -7,13 +7,16 @@
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory;
 import com.google.step.model.Deal;
+import java.util.ArrayList;
 
 public class DealManagerDatastore implements DealManager {
 
   private final DatastoreService datastore;
+  DealSearchManager searchManager;
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/datamanager/DealManagerDatastore.java]: private final

**Review by anuragshirolkar — DISMISSED** [review]

**Review by anuragshirolkar — APPROVED** [review]

**Pull request merged**